src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -260,6 +261,132 @@ public static void BuildChain_WithCertificatePolicy_NoMatch()
         }
 
         [Fact]
+        public static void BuildChain_MicrosoftDotCom_WithSystemRootCerts()
+        {
+            // Verifies that the microsoft.com certs build with only the certificates in the root store
+            // This is a canary test for BuildChain_MicrosoftDotCom_WithRootCertInUserAndSystemRootCertStores
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chainValidator = chainHolder.Chain;
+                bool chainBuildResult = chainValidator.Build(microsoftDotCom);
+
+                StringBuilder builder = new StringBuilder();
+                foreach (var status in chainValidator.ChainStatus)
+                {
+                    builder.AppendFormat("{0} {1}{2}", status.Status, status.StatusInformation, Environment.NewLine);
+                }
+
+                Assert.True(chainBuildResult, "Chain build for microsoft.com cert failed. Reason: " + builder.ToString());
+            }
+        }
+
+        [Fact]
+        [OuterLoop(/* Modifies user certificate store */)]
+        public static void BuildChain_MicrosoftDotCom_WithRootCertInUserAndSystemRootCertStores()
+        {
+            // Verfies that when the same root cert is placed in both a user and machine root certificate store, 
+            // any certs chain building to that root cert will build correctly
+            // 
+            // We use a copy of the microsoft.com SSL certs and root certs to validate that the chain can build 
+            // successfully
+
+            bool shouldInstallCertToUserStore = true;
+            bool installedCertToUserStore = false;
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+            {
+                // Check that microsoft.com's root certificate IS installed in the machine root store as a sanity step
+                using (var machineRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
+                {
+                    machineRootStore.Open(OpenFlags.ReadOnly);
+                    bool foundCert = false;
+
+                    foreach (var machineCert in machineRootStore.Certificates)
+                    {
+                        if (machineCert.Equals(microsoftDotComRoot))
+                        {
+                            foundCert = true;
+                        }
+
+                        machineCert.Dispose();
+
+                        if (foundCert)
+                        {
+                            break;
```

Comment:
Nope, can't break early; or it'll still send all the certs after this one to be finalized.  The Certificates property fully copies before returning.